crates/matrix-sdk-base/src/client.rs

```diff
@@ -539,7 +539,17 @@ impl BaseClient {
     ) {
         for raw_event in events {
             if let Ok(event) = raw_event.deserialize() {
-                changes.add_room_account_data(room_id, event, raw_event.clone());
+                changes.add_room_account_data(room_id, event.clone(), raw_event.clone());
+
+                if let AnyRoomAccountDataEvent::MarkedUnread(e) = event {
+                    if let Some(room) = changes.room_infos.get_mut(room_id) {
+                        room.base_info.is_marked_unread = e.content.unread;
+                    } else if let Some(room) = self.store.get_room(room_id) {
+                        let mut info = room.clone_info();
+                        info.base_info.is_marked_unread = e.content.unread;
+                        changes.add_room(info);
+                    }
```

Comment:
Please, can you explain why you look into `self.store` if `changes.room_infos` doesn't contain `room_id`?

Reply:
Changes contain processed joined rooms that came down in a sliding sync response. Rooms can also be updated from the extensions part of the response and those rooms might not be a part of the joined list. This ensures that if there's only room account data changes the room info will be updated correctly.